src/PostForm.tsx

```diff
@@ -0,0 +1,104 @@
+import usersFromServer from './api/users';
+import { useState } from 'react';
+import { getUserById } from './services/user';
+import { Todo } from './types/Todo';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const PostForm: React.FC<Props> = ({ onSubmit }) => {
+  const [titleErrorMessage, setTitleErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const hasTitleError = titleErrorMessage !== '';
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [userId, setUserId] = useState(0);
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setUserId(0);
+    setTitle('');
+    setTitleErrorMessage('');
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasUserIdError(!userId);
+
+    if (!title) {
+      setTitleErrorMessage('Please enter a title');
+
+      return;
+    }
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
```

Comment:
If `getUserById` is an asynchronous function, this code will not work as expected because it assumes `getUserById` returns a user object synchronously. Consider using `async/await` or handling the promise if `getUserById` is asynchronous.